### DISCUSSION TITLE
Nvidia support

### DISCUSSION THREAD

**Discussion by wierdkno1** [ravynsoft/ravynos#243] [General]
Will the property free bsd nvidia drivers 
Work with airyx os

**Comment by mszoek** [comment]
Unknown at this time. If they provide a kernel module for DRM/KMS, then likely yes. I'll look at what's provided if you can point me at a download link.

**Reply by mszoek** [reply]
Capturing this update for the record:
I tried setting up the proprietary nvidia drivers on my laptop which has dual Intel/nvidia GPUs. Although they do have a DRM module, it is missing necessary KMS support in the FreeBSD build.

**Reply by osenberg-x** [reply]
As me know, now, proprietary nvidia drivers support x11 only, this project will use wayland in the future, so mybe can not support proprietary nvidia drivers on airyx os

**Reply by mszoek** [reply]
It's true (and very unfortunate) that NVidia doesn't include crucial parts of the DRM/KMS code in their FreeBSD driver. There is some work here to use EGLStreams as a wayland DRM back-end that I am hoping to use: https://github.com/NVIDIA/egl-wayland. Mesa and wlroots (compositor library) both have some support for eglstreams.

PS: it's called "ravynOS" now :)

**Reply by osenberg-x** [reply]
I’m sorry, I made a mistake about os name. About supportion of graphic driver maybe ravynOS can abandon Nvidia until Nvidia changed their policy, like apple.  

Nvidia f**k you   —  Linus Tarvalds

**Reply by mszoek** [reply]
Apple has the luxury of making their own hardware, so they get to choose what components are used. I want this OS to be able to run on most machines, so intentionally ruling out a major vendor is not really an option. If we can't get it to work, that's different.

**Reply by jbeich** [reply]
See [nvidia-drm CFT thread](https://forums.freebsd.org/threads/call-for-testing-nvidia-drm-kernel-module.87161/). Both [EGL Streams](https://github.com/aritger/eglstreams-kms-example/issues/7) (egl-wayland) and GBM (mesa, nvidia-driver >= 495) require nvidia-drm. Beware, nvidia-drm appeared later on FreeBSD than Linux, so old GPUs relegated to legacy drivers are out of luck.
